python/cuda_parallel/tests/test_reduce_api.py

```diff
@@ -38,3 +40,17 @@ def op(a, b):
     # example-end reduce-min
     h_output = d_output.copy_to_host()
     assert h_output[0] == expected_output
+
+
+def test_device_reduce_dtype_mismatch():
+    dtypes = [numpy.int32, numpy.int64]
+    h_inits = [numpy.array([], dt) for dt in dtypes]
+    h_inputs = [numpy.array([], dt) for dt in dtypes]
+    d_outputs = [cuda.device_array(1, dt) for dt in dtypes]
+    d_inputs = [cuda.to_device(h_inp) for h_inp in h_inputs]
+
+    reduce_into = cudax.reduce_into(d_inputs[0], d_outputs[0], min_op, h_inits[0])
+
+    for ix in range(3):
+        with pytest.raises(TypeError, match=r"dtype mismatch: __init__=int32, __call__=int64"):
+          reduce_into(None, d_inputs[int(ix == 0)], d_outputs[int(ix == 1)], h_inits[int(ix == 2)])
```

Comment:
important: this file's purpose is testing code snippets in our docs. Let's put functionality tests into `test_reduce.py`.

Reply:
Done.